src/openslide-vendor-dicom.c

```
@@ -0,0 +1,943 @@
+  printf("  bot = %p\n", f->bot);
+}
+
+static void print_level(struct level *l) {
+  printf("level:\n" );
+  print_file(l->file);
+  printf("  base.downsample = %g\n", l->base.downsample);
+  printf("  grid = %p\n", l->grid);
+  printf("  format = %d\n", l->image_format);
+  printf("  image_width = %d\n", l->image_width);
+  printf("  image_height = %d\n", l->image_height);
+  printf("  tile_w = %d\n", l->tile_w);
+  printf("  tile_h = %d\n", l->tile_h);
+  printf("  num_frames = %d\n", l->num_frames);
+  printf("  tiles_across = %d\n", l->tiles_across);
+  printf("  tiles_down = %d\n", l->tiles_down);
+}
+
+static void print_frame(DcmFrame *frame) {
+  printf("value = %p\n", dcm_frame_get_value(frame));
+  printf("length = %u bytes\n", dcm_frame_get_length(frame));
+  printf("rows = %u\n", dcm_frame_get_rows(frame));
+  printf("columns = %u\n", dcm_frame_get_columns(frame));
+  printf("samples per pixel = %u\n",
+         dcm_frame_get_samples_per_pixel(frame));
+  printf("bits allocated = %u\n", dcm_frame_get_bits_allocated(frame));
+  printf("bits stored = %u\n", dcm_frame_get_bits_stored(frame));
+  printf("high bit = %u\n", dcm_frame_get_high_bit(frame));
+  printf("pixel representation = %u\n",
+         dcm_frame_get_pixel_representation(frame));
+  printf("planar configuration = %u\n",
+         dcm_frame_get_planar_configuration(frame));
+  printf("photometric interpretation = %s\n",
+         dcm_frame_get_photometric_interpretation(frame));
+  printf("transfer syntax uid = %s\n",
+         dcm_frame_get_transfer_syntax_uid(frame));
+}
+#endif /*DEBUG*/
+
+static bool dicom_detect(const char *filename,
+                         struct _openslide_tifflike *tl, 
+                         GError **err) {
+  // reject TIFFs
+  if (tl) {
+    g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED,
+                "Is a TIFF file");
+    return false;
+  }
+
+  // verify existence
+  GError *tmp_err = NULL;
+  if (!_openslide_fexists(filename, &tmp_err)) {
```

Is this necessary, or can we just let the open call fail if the file is missing?

---

OK, gone!